src/core/backbone_router/bbr_leader.cpp

```diff
@@ -102,71 +103,45 @@ void Leader::LogDomainPrefix(DomainPrefixState aState, const Ip6::Prefix &aPrefi
 
 const char *Leader::StateToString(State aState)
 {
-    const char *logString = "Unknown";
-
-    switch (aState)
-    {
-    case kStateNone:
-        logString = "None";
-        break;
-
-    case kStateAdded:
-        logString = "Added";
-        break;
-
-    case kStateRemoved:
-        logString = "Removed";
-        break;
-
-    case kStateToTriggerRereg:
-        logString = "Rereg triggered";
-        break;
-
-    case kStateRefreshed:
-        logString = "Refreshed";
-        break;
-
-    case kStateUnchanged:
-        logString = "Unchanged";
-        break;
-
-    default:
-        break;
-    }
-
-    return logString;
+    static const char *const kStateStrings[] = {
+        "None",            //  (0) kStateNone
+        "Added",           //  (1) kStateAdded
+        "Removed",         //  (2) kStateRemoved
+        "Rereg triggered", //  (3) kStateToTriggerRereg
+        "Refreshed",       //  (4) kStateRefreshed
+        "Unchanged",       //  (5) kStateUnchanged
+    };
+
+    static_assert(0 == kStateNone, "kStateNone value is incorrect");
+    static_assert(1 == kStateAdded, "kStateAdded value is incorrect");
+    static_assert(2 == kStateRemoved, "kStateRemoved value is incorrect");
+    static_assert(3 == kStateToTriggerRereg, "kStateToTriggerRereg value is incorrect");
+    static_assert(4 == kStateRefreshed, "kStateRefreshed value is incorrect");
+    static_assert(5 == kStateUnchanged, "kStateUnchanged value is incorrect");
+
+    return kStateStrings[aState];
```

Comment:
We don't handle larger-than-expected values here. How about defining a new enum value to indicate the end and have a static assert here to check if the end has the expected value?

Reply:
This can work, but I am in general not fond of adding an extra enum to the end to mark end of enumeration. It usually messes up the checks by toolchains/compilers in `switch()` statements (which check that all enum values are covered). We either need add this extra `enum` to the list of `case`s or add a `default` (which is not good since it kinda defeats the purpose of the toolchain checking it).